.travis.yml

```diff
@@ -30,17 +30,15 @@ jobs:
       name: "Bazel: //src_kt/... "
     - script: tools/run_in_docker "bazel --bazelrc=tools/travis.bazelrc build //src/... && bazel --bazelrc=tools/travis.bazelrc test //src/..."
       name: "Bazel: //src/..."
-    - script: tools/run_in_docker "bazel --bazelrc=tools/travis.bazelrc build //particles/... && bazel --bazelrc=tools/travis.bazelrc test //particles/..."
-      name: "Bazel: //particles/..."
+    - script: tools/run_in_docker "bazel --bazelrc=tools/travis.bazelrc build //particles/... --build_tag_filters kotlinc && bazel --bazelrc=tools/travis.bazelrc test //particles/... --build_tag_filters kotlinc  && tools/sigh testShells"
+      name: "Bazel: //particles/... (Kotlin) + testShells"
+    - script: tools/run_in_docker "bazel  --bazelrc=tools/travis.bazelrc build //particles/... --build_tag_filters emscripten && bazel --bazelrc=tools/travis.bazelrc test //particles/... --build_tag_filters emscripten && tools/sigh testShells"
+      name: "Bazel: //particles/... (C++) + testShells"
```

Comment:
Runing `testShells` twice is odd and the test results are not trustworthy if the two tests use different artifacts (iow, what Kotlin particles is the second one testing?). 

Sorry if I'm confused.

Reply:
Let me explain: I was worried that adding an extra step (testShells) would slow down our build. So I pulled out a trick to make the build go faster: 
- I added tags to our C++ and Kotlin Wasm builds
- I filter the bazel build / test process looking by Wasm compiler type

The value of this is that each VM won't download two compiler toolchains, it'll only download the one needed for the language. 

This means that we test the Kotlin particles and C++ particles in shells twice. In each case, one language's Wasm artifacts are stale, and the others have been freshly compiled.

Reply:
_This means that we test the Kotlin particles and C++ particles in shells twice. In each case, one language's Wasm artifacts are stale, and the others have been freshly compiled_

Seems like this is exactly what I was fearing. Running tests twice to access a subset each time is bad. Stale is bad.